activerecord/test/cases/primary_class_test.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+
+class PrimaryClassTest < ActiveRecord::TestCase
+  self.use_transactional_tests = false
+
+  def teardown
+    clean_up_connection_handler
+  end
+
+  class PrimaryAppRecord < ActiveRecord::Base
```

Comment:
@eileencodes @rafaelfranca 

I found these tests to be broken after an attempt to fix the bug because it introduces the inheritance of an abstract class from a non-abstract class, while it seems it should not be possible as it causes a confusing behaviour of `.base_class` and all the low level things depending on it. 

See https://github.com/rails/rails/blob/main/activerecord/test/cases/inheritance_test.rb#L147

Originally from  https://github.com/rails/rails/commit/8ffb22056c2896a67917b813064b2247233d3a08#diff-6e5593e0695d9011d63ec92113658dfd5fd2661a82380aa3410da270bbeffd8bR1407

I think it should be fixed like this:

``` ruby
class PrimaryAppRecord < ActiveRecord::Base
  self.abstract_class = true
end
```

Do you mind a change?

Reply:
I don't quite follow but you can open an issue or PR and we can discuss there.